src/core/core.tooltip.js

```diff
@@ -604,11 +608,18 @@ module.exports = function(Chart) {
 			}
 
 			ctx.fillStyle = mergeOpacity(vm.backgroundColor, opacity);
+			ctx.lineWidth = vm.borderWidth;
+			ctx.strokeStyle = vm.borderColor;
+
 			ctx.beginPath();
 			ctx.moveTo(x1, y1);
 			ctx.lineTo(x2, y2);
 			ctx.lineTo(x3, y3);
+			ctx.stroke();
+			ctx.fill();
 			ctx.closePath();
+
+			helpers.drawRoundedRectangle(ctx, ptX, ptY, size.width, size.height, vm.cornerRadius);
```

Comment:
does this end up drawing the background twice? My concern is that because the background is drawn twice this could cause performance issues when the tooltip updates frequently. Would it make sense to refactor `drawBackground` and `drawCaret` to draw the border only once?